modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/exception/CompositeClassNotFoundException.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.classloader.exception;
+
+import static com.google.common.collect.ImmutableList.copyOf;
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+
+import org.mule.runtime.module.artifact.classloader.ClassLoaderLookupStrategy;
+
+import java.util.List;
+
+/**
+ * Extends {@link ClassNotFoundException}, composing the individual exceptions of each place where
+ * the class was looked for and wasn't found.
+ */
+public class CompositeClassNotFoundException extends ClassNotFoundException
```

Comment:
We would need some test for this new feature. Not sure if that is doable, but having a feature that has no test and won't be activated by default  makes it prone to problems. Can you review if is possible to add tests ensuring that the right exceptions are gathered and the right messages created?